### PR TITLE
Add KB09 keyboard

### PR DESCRIPTION
**Pull request by DOIO2022**
## Description

Add KB09 keyboard

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/doio/kb09/info.json [hunk 1]
```diff
@@ -0,0 +1,126 @@
+{
+    "manufacturer": "DOIO",
+    "keyboard_name": "KB09-01",
+    "maintainer": "DOIO2022",
+    "bootloader": "stm32duino",
+    "diode_direction": "COL2ROW",
+    "dynamic_keymap": {
+        "layer_count": 4
+    },
```
Comment (drashna) [inline keyboards/doio/kb09/info.json:9]: This is not needed, as 4 is the default value for this. 
```suggestion
```

File: keyboards/doio/kb09/info.json [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+    "encoder": {
+        "rotary": [
+            {"pin_a": "B5", "pin_b": "B6"},
+            {"pin_a": "A1", "pin_b": "A2"}
+        ]
+    },
+    "features": {
+        "bootmagic": true,
+        "command": false,
+        "console": false,
```
Comment (drashna) [inline keyboards/doio/kb09/info.json:19]:
```suggestion
```

File: keyboards/doio/kb09/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+#include QMK_KEYBOARD_H
+
+enum layer_names {
+    _LAY0,
+    _LAY1,
+    _LAY2,
+    _LAY3
+};
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
```
Comment (drashna) [inline keyboards/doio/kb09/keymaps/default/keymap.c:28]: please change the tabs to spaces.

File: keyboards/doio/kb09/keymaps/via/keymap.c [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+#include QMK_KEYBOARD_H
+
+enum layer_names {
+    _LAY0,
+    _LAY1,
+    _LAY2,
+    _LAY3
+};
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
```
Comment (drashna) [inline keyboards/doio/kb09/keymaps/via/keymap.c:28]: same

**Review by drashna — CHANGES_REQUESTED** [review]
Is there a reason for the removed keyboards? If not, those changes should be reverted.

File: keyboards/doio/kb09/keymaps/default/keymap.c (hunk 3 above)
Comment (DOIO2022) [reply keyboards/doio/kb09/keymaps/default/keymap.c:28]: I've changed it .  Thank you!

File: keyboards/doio/kb09/keymaps/via/keymap.c (hunk 4 above)
Comment (DOIO2022) [reply keyboards/doio/kb09/keymaps/via/keymap.c:28]: I've changed it .  Thank you!

**Comment by DOIO2022** [comment]
> Is there a reason for the removed keyboards? If not, those changes should be reverted.

I made an operation error and only partially reverted, so I manually reverted.

File: keyboards/doio/kb09/keymaps/via/keymap.c (hunk 4 above)
Comment (DOIO2022) [reply keyboards/doio/kb09/keymaps/via/keymap.c:28]: @drashna  HI,please review my modified content.Thank you!

**Review by zvecr — CHANGES_REQUESTED** [review]
* Changes to other keyboards **must** be reverted
  * This has previously been requested and is still outstanding
  * `kb38` changes should be submitted separately

**Comment by DOIO2022** [comment]
> * Changes to other keyboards **must** be reverted
>   
>   * This has previously been requested and is still outstanding
>   * `kb38` changes should be submitted separately

Hi , I've reverted the  kb38 info.json . Please review again， Thank you！

**Comment by zvecr** [comment]
There are other files that are modified and **still** need to be reverted. This can easily be shown on the "Files Changed" tab.

**Comment by DOIO2022** [comment]
> There are other files that are modified and **still** need to be reverted. This can easily be shown on the "Files Changed" tab.

QQ截图20240620192739
I committed these files on the browser, can you tell me how I revert these , I can't find the revert button.  Thank you  so much!

**Pull request closed without merging**